shared/reducers/fs.js

```diff
@@ -170,8 +170,20 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
         console.warn(`bad parentPath: ${parentPathItem.type}`)
         return state
       }
+
+      const existingNewFolderNames = new Set(
+        state.edits
+          .valueSeq()
```

Comment:
Maybe faster to filter first then map then use toSet to get the immutable set vs the js set